src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1007,23 +1075,41 @@ query_path(const char *szInput, int *is_target_path, dfs_obj_t **parent, char *i
 		/* trying to avoid lock as much as possible */
 		if (!daos_inited) {
 			/* daos_init() is expensive to call. We call it only when necessary. */
+
+			D_MUTEX_LOCK(&lock_global);
```

Comment:
why is this needed for the entire block? isn't this just needed internally for consume_low_fd()?

Reply:
You are right. I will move the lock inside consume_low_fd(). Thank you!

Reply:
well my point was just that if you already added a lock on top (not sure if that one is needed), do you need this one too?

Reply:
actually nevemind that comment was for the next lock not the lock_global